### PR TITLE
Ct 1785 edit ico closure

### PR DESCRIPTION
**Pull request by beccapearce**

### CODE REVIEW

File: app/controllers/cases_controller.rb [hunk 1]
```diff
@@ -234,6 +234,9 @@ def edit
 
   def edit_closure
     authorize @case, :update_closure?
+    if @case.ico?
+      @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'ICO_decision')
```
Comment (aliuk2012) [inline app/controllers/cases_controller.rb]: I think `ICO_decision` should be lower case to match the directory that is used in s3 bucket

File: app/controllers/cases_controller.rb [hunk 2]
```diff
@@ -363,6 +366,9 @@ def update_closure
 
     @case.prepare_for_close
     close_params = process_closure_params(@case.type_abbreviation)
+    if @case.ico?
+      @s3_direct_post = S3Uploader.s3_direct_post_for_case(@case, 'responses')
```
Comment (aliuk2012) [inline app/controllers/cases_controller.rb]: could you change `responses` to `ico_decisions`

File: config/state_machine/configs/00_moj/30_ico/20_ico_trigger_workflow.yml [hunk 3]
```diff
@@ -66,7 +66,6 @@
                 link_a_case:
                 remove_linked_case:
                 update_closure:
-                  if: Workflows::Predicates#can_edit_closure
```
Comment (aliuk2012) [inline config/state_machine/configs/00_moj/30_ico/20_ico_trigger_workflow.yml]: why are these removed?

File: config/state_machine/moj.yml [hunk 4]
```diff
@@ -1309,7 +1309,6 @@ case_types:
                 link_a_case:
                 remove_linked_case:
                 update_closure:
-                  if: Workflows::Predicates#can_edit_closure
```
Comment (aliuk2012) [inline config/state_machine/moj.yml]: why are these removed?

File: app/controllers/cases_controller.rb (hunk 1 above)
Comment (beccapearce) [reply app/controllers/cases_controller.rb]: Yep, still working on this PR

File: app/controllers/cases_controller.rb (hunk 2 above)
Comment (beccapearce) [reply app/controllers/cases_controller.rb]: yep

File: config/state_machine/configs/00_moj/30_ico/20_ico_trigger_workflow.yml (hunk 3 above)
Comment (beccapearce) [reply config/state_machine/configs/00_moj/30_ico/20_ico_trigger_workflow.yml]: It's not needed on ICOs

File: config/state_machine/moj.yml (hunk 4 above)
Comment (beccapearce) [reply config/state_machine/moj.yml]: It's not needed on ICOs

**Review by aliuk2012 — DISMISSED** [review]
This PR seems to introduce `config/state_machine/moj.yml`

File: spec/support/features/steps/cases/edit_steps.rb [hunk 5]
```diff
@@ -186,8 +186,16 @@ def edit_ico_case_closure_step(kase:, decision_received_date: Date.today, ico_de
   expect(cases_show_page).to be_displayed(id: kase.id)
   expect(cases_show_page.notice.text)
     .to eq 'You have updated the closure details for this case.'
-  # expect(cases_show_page.case_details.response_details.whatever_al_calls_this_field.data.text)
+  # uncomment when this section is added to the page
+  # expect(cases_show_page.ico.case_details.response_details.date_ico_decision_received.data.text)
```
Comment (aliuk2012) [inline spec/support/features/steps/cases/edit_steps.rb]: This is not in the current designs and the only place this is played back is further up the page where it shows what the final decision was and the date it was received "ICO decision section"

So we might as well remove this commented out text

Comment (beccapearce) [reply spec/support/features/steps/cases/edit_steps.rb]: Fair enough, I added it in though because I saw it on one of the designs.

File: app/views/cases/ico/_close_form.html.slim [hunk 6]
```diff
@@ -10,6 +10,8 @@
             form_hint_text: t('.date_example'),
             today_button: {class: ''} }
 
+  = render partial: 'cases/ico/ico_final_decision', locals: {case_details: @case}
```
Comment (misaka) [inline app/views/cases/ico/_close_form.html.slim]: The partial used here actually renders something different from what's on the designs. The design for what it renders ([from the "ICO Appeal case details closed overturned" design](https://jmwvac.axshare.com/#g=1&p=ico_appeal_case_details_closed_overturned)):

image

But the design for the [ICO Appeal case details edit ICO decision design](https://jmwvac.axshare.com/#g=1&p=ico_appeal_case_details_edit_ico_decision) looks like:

image

So I think we either need to sync these up, or look at a different design for this page. I think @aliuk2012 is working on this as part of CT-1848 so may be worth checking in with him.

**Review by misaka — CHANGES_REQUESTED** [review]

File: config/state_machine/configs/00_moj/30_ico/20_ico_trigger_workflow.yml (hunk 3 above)
Comment (misaka) [reply config/state_machine/configs/00_moj/30_ico/20_ico_trigger_workflow.yml]: :100:

File: app/views/cases/ico/_close_form.html.slim (hunk 6 above)
Comment (beccapearce) [reply app/views/cases/ico/_close_form.html.slim]: Cool, I'll ask Al

File: spec/support/features/steps/cases/edit_steps.rb [hunk 7]
```diff
@@ -151,3 +151,51 @@ def edit_sar_case_closure_step(kase:, date_responded: Date.today, tmm: false) #
...
+
+  if kase.ico_decision == 'upheld'
+    expect(cases_edit_closure_page.ico_decision.upheld).to be_checked
+    expect(cases_edit_closure_page.ico_decision.overturned ).not_to be_checked
+  elsif kase.ico_decision == 'overturned'
+    expect(cases_edit_closure_page.ico_decision.upheld).not_to be_checked
+    expect(cases_edit_closure_page.ico_decision.overturned ).to be_checked
+  end
+
+  cases_edit_closure_page.fill_in_ico_date_responded(decision_received_date)
```
[the remark this answers is not in the document]
Comment (beccapearce) [reply spec/support/features/steps/cases/edit_steps.rb]: yes, but it isn't displayed on the page so I don't think it belongs here, I'll add something to the controller test

[the remark this answers is not in the document]
Comment (beccapearce) [reply spec/support/features/steps/cases/edit_steps.rb]: actually it's already tested in the controller https://github.com/ministryofjustice/correspondence_tool_staff/pull/407/files#diff-573694c64e695c97b5ea96166ad584daR254

**Comment by misaka** [comment]
We are waiting for @aliuk2012 and Mia to work out the case attachments stuff to see if we need to change the "Original ICO's decision document" section here.

File: app/views/cases/ico/_close_form.html.slim (hunk 6 above)
Comment (misaka) [reply app/views/cases/ico/_close_form.html.slim]: We've decided to defer fixing the "ICO's decision" section for another ticket. @aliuk2012 and Mia are working on standardising the way they look.

**Review by misaka — APPROVED** [review]
